File: BubbleSort.py
list = [1,2,3,4,5]

def bubbleSort(list,n):
    last = n-1
    for i in range(0,n):
        exchs = 0
        for j in range(0,n-1):
            if(list[j] > list[j+1]):
                list[j],list[j+1] = list[j+1],list[j]
                exchs += 1
        if(exchs == 0):
            print("Vector is sorted!!")
            return last
        else:
            last = last -1
    return       

last = bubbleSort(list,len(list))
print("Sorted list using bubblesort is : ",list)
print("Last exchange at :",last)

File: BubbleSort.java
public class BubbleSort {
    public static void main(String[] args) {
        int arr[] = new int[]{51,238,243,1,9};
       
        for(int i=0;i<=(arr.length-1);i++){
            for(int j=0;j<=(arr.length-2);j++){
                if(arr[j]>arr[j+1]){
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }
        System.out.println("the arr is");  for(int i=0;i<arr.length;i++){
            
        System.out.println(arr[i]);
        
        }
    }
}
